### PR TITLE
Move some methods from `tcx.hir()` to `tcx`

### PR DESCRIPTION
**Pull request by zetanumbers**
https://github.com/rust-lang/rust/pull/118256#issuecomment-1826442834

Renamed:
- find -> opt_hir_node
- get -> hir_node
- find_by_def_id -> opt_hir_node_by_def_id
- get_by_def_id -> hir_node_by_def_id

### CODE REVIEW

File: compiler/rustc_middle/src/hir/map/mod.rs [hunk 1]
```diff
@@ -128,6 +128,40 @@ impl<'hir> Iterator for ParentOwnerIterator<'hir> {
     }
 }
```
Comment (petrochenkov) [inline compiler/rustc_middle/src/hir/map/mod.rs:129]: Could you add a FIXME comment to `struct Map` above, saying that the structure was necessary in the past but now it only serves as "namespace" for HIR-related methods, and can be removed if all the methods are reasonably renamed and moved to `tcx` (https://github.com/rust-lang/rust/pull/118256#issuecomment-1826442834).

File: compiler/rustc_middle/src/hir/map/mod.rs [hunk 2]
```diff
@@ -128,6 +128,40 @@ impl<'hir> Iterator for ParentOwnerIterator<'hir> {
...
+    /// Retrieves the `hir::Node` corresponding to `id`, returning `None` if cannot be found.
+    #[inline]
+    pub fn opt_hir_node_by_def_id(self, id: LocalDefId) -> Option<Node<'tcx>> {
+        self.opt_hir_node(self.opt_local_def_id_to_hir_id(id)?)
+    }
+
+    /// Retrieves the `hir::Node` corresponding to `id`, panicking if it cannot be found.
+    #[track_caller]
+    pub fn hir_node(self, id: HirId) -> Node<'tcx> {
+        self.opt_hir_node(id).unwrap_or_else(|| bug!("couldn't find hir id {} in the HIR map", id))
```
Comment (petrochenkov) [inline compiler/rustc_middle/src/hir/map/mod.rs:153]:
```suggestion
        self.opt_hir_node(id).unwrap_or_else(|| bug!("couldn't find HIR node for hir id {id:?}"))
```

File: compiler/rustc_middle/src/hir/map/mod.rs [hunk 3]
```diff
@@ -128,6 +128,40 @@ impl<'hir> Iterator for ParentOwnerIterator<'hir> {
...
+    }
+
+    /// Retrieves the `hir::Node` corresponding to `id`, panicking if it cannot be found.
+    #[inline]
+    #[track_caller]
+    pub fn hir_node_by_def_id(self, id: LocalDefId) -> Node<'tcx> {
+        self.opt_hir_node_by_def_id(id)
+            .unwrap_or_else(|| bug!("couldn't find {:?} in the HIR map", id))
```
Comment (petrochenkov) [inline compiler/rustc_middle/src/hir/map/mod.rs:161]:
```suggestion
            .unwrap_or_else(|| bug!("couldn't find HIR node for def id {id:?}"))
```

**Comment by petrochenkov** [comment]
Are there any cases where
```rust
let hir_id = tcx.(opt_)local_def_id_to_hir_id(def_id);
tcx.(opt_)hir_node(hir_id);
```
could be replaced with
```rust
tcx.(opt_)hir_node_by_def_id(def_id);
```
?

(I.e. where `hir_id` is only evaluated to be passed to `hir_node`.)

**Comment by ZetaNumbers** [comment]
Oh no, I'm so sorry for pinging everyone! It was my error with git, I've fixed it. I'm sorry!

**Comment by ZetaNumbers** [comment]
Is there a way to reset the ping or delete rustbot's message?

**Comment by ouz-a** [comment]
Don't worry about it this kind of accidents do happen 😄

File: compiler/rustc_middle/src/hir/map/mod.rs (hunk 1 above)
Comment (ZetaNumbers) [reply compiler/rustc_middle/src/hir/map/mod.rs:133]: Added in 9509ff4d8089c879efb75accb409fa331a6804fe

**Comment by petrochenkov** [comment]
r=me after squashing commits and making CI green.

**Comment by ZetaNumbers** [comment]
All set and done

**Pull request merged**